raster/r.kappa/testsuite/test_r_kappa.py

```diff
@@ -292,5 +297,194 @@ def test_standard_output(self):
+            output=cls.clas[1],
+            quiet=True,
+        )
+        cls.outp.append(
+            {
+                "map1": cls.refs[1],
+                "map2": cls.clas[1],
+                "observations": 25,
+                "correct": 0,
+                "total_acc": 0.0,
+                "kappa": 0.0,
+                "kappa_var": 0.0,
+                "cats": [0, 1, 2, 3, 4, 9],
+                "matrix": [
+                    [0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0],
+                    [0, 0, 0, 0, 0, 0],
+                    [8, 8, 4, 1, 4, 0],
+                ],
+                "prod_acc": [0.0, 0.0, 0.0, 0.0, 0.0, -999],
+                "user_acc": [-999, -999, -999, -999, -999, 0.0],
+                "cond_kappa": [-999, -999, -999, -999, -999, 0.0],
+            }
+        )
+
+        # Degenerate case #1
+        cls.refs.append(tempname(10))
+        cls.clas.append(tempname(10))
+        cls.runModule(
+            "r.mapcalc",
+            expression=f"{cls.refs[2]}=null()",
+            quiet=True,
+        )
+        cls.runModule(
+            "r.mapcalc",
+            expression=f"{cls.clas[2]}=null()",
+            quiet=True,
+        )
+        cls.outp.append(
+            {
+                "map1": cls.refs[2],
+                "map2": cls.clas[2],
+                "observations": 0,
+                "correct": 0,
+                "total_acc": 0.0,
+                "kappa": -999.0,
+                "kappa_var": -999.0,
+                "cats": [],
+                "matrix": [[]],
+                "prod_acc": [],
+                "user_acc": [],
+                "cond_kappa": [],
+            }
+        )
+
+        # Degenerate case #2
+        cls.refs.append(tempname(10))
+        cls.clas.append(tempname(10))
+        cls.runModule(
+            "r.mapcalc",
+            expression=f"{cls.refs[3]}=1",
+            quiet=True,
+        )
+        cls.runModule(
+            "r.mapcalc",
+            expression=f"{cls.clas[3]}=null()",
+            quiet=True,
+        )
+        cls.outp.append(
+            {
+                "map1": cls.refs[3],
+                "map2": cls.clas[3],
+                "observations": 0,
+                "correct": 0,
+                "total_acc": 0.0,
+                "kappa": -999.0,
+                "kappa_var": -999.0,
+                "cats": [],
+                "matrix": [[]],
+                "prod_acc": [],
+                "user_acc": [],
+                "cond_kappa": [],
+            }
+        )
+
+    @classmethod
+    def tearDownClass(cls):
+        """Remove temporary data"""
+        cls.del_temp_region()
+        for ref in cls.refs:
+            cls.runModule("g.remove", flags="f", type="raster", name=ref)
+        for clas in cls.clas:
+            cls.runModule("g.remove", flags="f", type="raster", name=clas)
```

Comment:
With whole words, this can nicely work with singulars and plurals as `for reference in references` and `for classification in classifications`.

Reply:
Come on, this code saves a few bytes of disk space! Besides, don't you love keeping as close to Fortran as possible? ;-)